src/client/input.js

```diff
@@ -51,12 +30,34 @@ export default class Input extends Component {
 		wrapperClassName: PropTypes.string
 	};
 
+	static defaultProps = {
+		addonAfter: null,
+		addonBefore: null,
+		bsSize: null,
+		buttonAfter: null,
+		buttonBefore: null,
+		children: null,
+		groupClassName: null,
+		hasFeedback: null,
+		help: null,
+		id: null,
+		label: null,
+		labelClassName: null,
+		name: null,
+		standalone: false,
+		tooltipText: null,
+		type: null,
+		validationState: null,
+		wrapperClassName: null
+	};
+
 	constructor(props, context) {
 		super(props, context);
 
 		this.refFormControl = null;
 	}
 
+	// eslint-disable-next-line accessor-pairs
```

Comment:
Agreed. In fact, this Input component was only ever meant to be temporary. react-bootstrap removed their own Input component and replaced it with a building-block approach, so when we updated, we needed to re-create it as a wrapper around those building blocks. The idea was eventually that occurences might be replaced by bespoke combinations of building blocks, cutting away unnecessary logic/compatability code inside the Input class.

It might even be worth evaluating whether react-bootstrap is still the best option, or whether reactstrap or just using bootstrap directly would be better in some cases...